File: packages/android/src/main/java/com/kozmos/components/ScrollArea/ScrollArea.figma.kt
package com.kozmos.components.scrollarea

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=655-4973")
class KozmosScrollAreaConnect {
    @FigmaProperty(FigmaType.Enum, "Orientation")
    val orientation: KozmosScrollAreaOrientation = Figma.mapping(
        "Vertical" to KozmosScrollAreaOrientation.Vertical,
        "Horizontal" to KozmosScrollAreaOrientation.Horizontal,
        "Both" to KozmosScrollAreaOrientation.Both
    )

    @Composable
    fun ComponentExample() {
        KozmosScrollArea(orientation = orientation) {
            Text("Scrollable content")
        }
    }
}

File: packages/ios/Sources/Components/ScrollArea/ScrollArea.figma.swift
import SwiftUI
import Figma

struct KozmosScrollAreaConnect: FigmaConnect {
    let component = KozmosScrollArea<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=655-4973"

    @FigmaEnum(
        "Orientation",
        mapping: [
            "Vertical": KozmosScrollAreaOrientation.vertical,
            "Horizontal": KozmosScrollAreaOrientation.horizontal,
            "Both": KozmosScrollAreaOrientation.both
        ]
    )
    var orientation: KozmosScrollAreaOrientation = .vertical

    @FigmaEnum(
        "Scrollbar",
        mapping: [
            "Hidden": false,
            "Visible": true
        ]
    )
    var showsIndicators: Bool = false

    var body: some View {
        KozmosScrollArea(orientation: self.orientation, showsIndicators: self.showsIndicators) {
            AnyView(
                VStack(alignment: .leading, spacing: KozmosDimensions.primitivesLayoutSpacing200) {
                    Text("Scrollable content")
                    Text("Content continues")
                }
            )
        }
    }
}
